src/cargo/ops/cargo_remove.rs

```diff
@@ -50,6 +64,23 @@ pub fn remove(options: &RemoveOptions<'_>) -> CargoResult<()> {
         // crate, then we need to drop any explicitly activated features on
         // that crate.
         manifest.gc_dep(dep);
+
+        // If this was the last instance of a workspace dependency, remove it
+        // from the workspace dependencies.
+        if is_workspace_dep {
+            let workspace_manifest = match &mut virtual_manifest {
+                Some(virtual_manifest) => virtual_manifest,
+                None => &mut manifest.data,
```

Comment:
This removes it from the workspace if its the workspace root but what if its not the workspace root and not a virtual workspace?